functions/webhooks/twitter/TwitterToFlex.ts

```diff
@@ -179,33 +209,33 @@ const sendChatMessage = async (
 const sendMessageToFlex = async (
   context: Context<EnvVars>,
   senderId: string,
-  senderName: string,
   senderScreenName: string,
   messageText: string,
   forUserId: string,
 ) => {
   const uniqueChannelName = `${twitterUniqueNamePrefix}${senderId}`;
   const uniqueSenderName = `${twitterUniqueNamePrefix}${senderId}`;
-  let channelSid;
+  let channelSid: string | undefined;
 
   try {
-    const channel = await retrieveChannel(context, uniqueChannelName);
-    channelSid = channel.sid;
-  } catch (err) {
-    try {
+    channelSid = await retrieveChannel(context, uniqueSenderName);
+
+    if (!channelSid) {
       console.log('Creating new channel');
-      const newChannel = await createTwitterChannel(
+      const newChannelSid = await createTwitterChannel(
         context,
         uniqueChannelName,
         senderId,
         uniqueSenderName,
         senderScreenName,
         forUserId,
       );
-      channelSid = newChannel.sid;
-    } catch (err2) {
-      throw new Error(`Error while creating the new channel ${err2.message}`);
+
+      channelSid = newChannelSid;
+      await createUserChannelMap(context, uniqueSenderName, channelSid);
     }
+  } catch (err) {
+    throw new Error(`Error while creating the new channel ${err.message}`);
   }
```

Comment:
If for some reason `createUserChannelMap` fails, we will have a "hanging" channel (that was just created above). Is that a problem? Should we try to remove this "hanging" channel on the catch/finally block?

Reply:
I don't think there is any problem but already added a cleanup for it.